lib/sles4sap/console_redirection.pm

```diff
@@ -123,32 +127,34 @@ sub connect_target_to_serial {
     my $redirect_opts = "-R $redirect_port:$redirect_ip:$redirect_port";
     enter_cmd "ssh $ssh_opt $redirect_opts $args{ssh_user}\@$args{destination_ip} 2>&1 | tee -a /dev/$serialdev";
     handle_login_prompt($args{ssh_user});
-    die 'Failed redirecting console' unless check_serial_redirection();
+
+    my $redirection_active = check_serial_redirection();
+    if ($args{fail_ok} && !$redirection_active) {
+        record_info('Redirect FAIL', 'Redirection FAILED. Argument "fail_ok" is set, test will continue.');
+        return 0;
+    }
+    die 'Failed redirecting console' unless $redirection_active;
     record_info('Redirect ON', "Serial redirection established to: $args{destination_ip}");
+    return 1;
 }
 
 =head2 disconnect_target_from_serial
 
-    disconnect_target_from_serial( [, base_vm_machine_id=$base_vm_machine_id]);
-
-B<base_vm_machine_id>: ID of the base VM before redirection. Default is BASE_VM_ID value set by redirect_init()
+    disconnect_target_from_serial();
 
 Disconnects target from serial console by typing 'exit' command until host machine ID matches ID of the worker VM.
 
 =cut
 
 sub disconnect_target_from_serial {
-    my (%args) = @_;
-    $args{base_vm_machine_id} //= get_required_var('BASE_VM_ID');
     set_serial_term_prompt();
-    my $serial_redirection_status = check_serial_redirection(base_vm_machine_id => $args{base_vm_machine_id});
+    my $serial_redirection_status = check_serial_redirection();
     while ($serial_redirection_status != 0) {
         enter_cmd('exit');    # Enter command and wait for screen start changing
         $testapi::distri->{serial_term_prompt} = '';    # reset console prompt
         wait_serial(qr/Connection.*closed./, timeout => 10, quiet => 1);    # Wait for connection to close
         wait_serial(qr/# |> /, timeout => 10, quiet => 1);    # Wait for console prompt to appear
-        set_serial_term_prompt();    # after logout user might change and prompt with it.
```

Comment:
`set_serial_term_prompt()` is checked right at the beginning of `check_redirection_status`. No need to pollute the logs